### ISSUE TITLE
bug report: Mail forwarding bounced by gmail

### ISSUE THREAD

**Issue by tzerber** [docker-mailserver/docker-mailserver#3642]
### 📝 Preliminary Checks

- [X] I tried searching for an existing issue and followed the [debugging docs](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/) advice, but still need assistance.

### 👀 What Happened?

Hi all,
I'm using one single instance of docker mailserver to handle several domains. One of the domains is a pure forwarder, i.e. user@domain is forwarded to user@gmail and so on. All worked fine for more than a year, but last month I started getting emails bouncing from the forwarder. The sender gets the following: 
`host gmail-smtp-in.l.google.com[142.250.27.27] said:
550-5.7.26 This mail has been blocked because the sender is
unauthenticated. 550-5.7.26 Gmail requires all senders to authenticate with
either SPF or DKIM. 550-5.7.26 550-5.7.26 Authentication results:
550-5.7.26 DKIM = did not pass 550-5.7.26 SPF [sender-domain.com] with ip:
[my-mail-server-ip] = did not pass 550-5.7.26 550-5.7.26 To mitigate this
issue, please visit Gmail's authentication guide 550-5.7.26 for
instructions on setting up authentication: 550 5.7.26`

For some unknown reason, when forwarding the mail my mailserver is trying to present itself as the sender's domain. The sender's domain is properly set up (dkim,spf etc) and my server is also properly set up (dmarc dkim spf etc). Email traffic there is roughly 20 emails per day for the whole domain, most of them from postmaster's cron. 

### 👟 Reproduction Steps

set up a forwarder with
`./setup.sh alias add test123@domain.com user@gmail.com`
then send an email to test123@domain.com from something different than gmail, in my case using latest thunderbird and a paid email hosting company

you will get the output (pasted below) and a email informing you that the email bounced (like the text above). I have snipped private domain names and IPs for obvious reasons.

### 🐋 DMS Version

v12.1.0

### 💻 Operating System and Architecture

Ubuntu 20.04.6 LTS

### ⚙️ Container configuration files

```yml
mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    hostname: mail
    domainname: my-domain.com
    container_name: mailserver
    ports:
      - "25:25"
      - "143:143"
      - "587:587"
      - "993:993"
    logging:
      driver: "json-file"
      options:
        max-size: "10m"
    volumes:
      - maildata:/var/mail
      - mailstate:/var/mail-state
      - maillogs:/var/log/mail
      - proxy_certs:/etc/letsencrypt/live
      - ./config/:/tmp/docker-mailserver/
      - ./cron/sa-learn:/etc/cron.d/sa-learn
```

### 📜 Relevant log output

```Text
mailserver                | Nov 18 10:59:17 mail postfix/postscreen[177010]: CONNECT from [23.83.216.34]:23397 to [172.20.0.3]:25
mailserver                | Nov 18 10:59:23 mail postfix/postscreen[177010]: PASS NEW [23.83.216.34]:23397
mailserver                | Nov 18 10:59:23 mail postfix/smtpd[177011]: connect from cheetah.pear.relay.mailchannels.net[23.83.216.34]
mailserver                | Nov 18 10:59:24 mail postfix/smtpd[177011]: Anonymous TLS connection established from cheetah.pear.relay.mailchannels.net[23.83.216.34]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest SHA256
mailserver                | Nov 18 10:59:25 mail policyd-spf[177228]: prepend Received-SPF: Pass (mailfrom) identity=mailfrom; client-ip=23.83.216.34; helo=cheetah.pear.relay.mailchannels.net; envelope-from=sender@sender-domain; receiver=<UNKNOWN>
mailserver                | Nov 18 10:59:25 mail dovecot: auth: passwd-file(test@my-domain): unknown user
mailserver                | Nov 18 10:59:25 mail dovecot: auth: passwd-file(test@my-domain): unknown user
mailserver                | Nov 18 10:59:25 mail dovecot: auth: passwd-file(test@my-domain): unknown user
mailserver                | Nov 18 10:59:25 mail dovecot: auth: passwd-file(test@my-domain): unknown user
mailserver                | Nov 18 10:59:25 mail dovecot: auth: passwd-file(test@my-domain): unknown user
mailserver                | Nov 18 10:59:25 mail dovecot: auth: passwd-file(test@my-domain): unknown user
mailserver                | Nov 18 10:59:25 mail postfix/smtpd[177011]: B0E6F5828BC: client=cheetah.pear.relay.mailchannels.net[23.83.216.34]
mailserver                | Nov 18 10:59:26 mail postfix/cleanup[177233]: B0E6F5828BC: message-id=<20c59ac9-9afc-46bc-afc9-1b4c7763e612@sender-domain>
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: cheetah.pear.relay.mailchannels.net [23.83.216.34] not internal
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: not authenticated
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: DKIM verification successful
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: s=default d=sender-domain a=rsa-sha256 SSL
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC: ignoring invalid ARC-Authentication-Results header "i=1;#012#011rspamd-55bcb54c45-c8qs5;#012#011auth=pass smtp.auth=thundermail smtp.mailfrom=sender@sender-domain"
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC ignoring Authentication-Results at 20 from cloud104.unlimitedwebhosting.co.uk
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC: sender-domain none
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: cheetah.pear.relay.mailchannels.net [23.83.216.34] not internal
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: not authenticated
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: DKIM verification successful
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: s=default d=sender-domain a=rsa-sha256 SSL
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC: ignoring invalid ARC-Authentication-Results header "i=1;#012#011rspamd-55bcb54c45-c8qs5;#012#011auth=pass smtp.auth=thundermail smtp.mailfrom=sender@sender-domain"
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC ignoring Authentication-Results at 19 from cloud104.unlimitedwebhosting.co.uk
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC: sender-domain none
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: cheetah.pear.relay.mailchannels.net [23.83.216.34] not internal
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: not authenticated
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: DKIM verification successful
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: s=default d=sender-domain a=rsa-sha256 SSL
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC: ignoring invalid ARC-Authentication-Results header "i=1;#012#011rspamd-55bcb54c45-c8qs5;#012#011auth=pass smtp.auth=thundermail smtp.mailfrom=sender@sender-domain"
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC ignoring Authentication-Results at 19 from cloud104.unlimitedwebhosting.co.uk
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC: sender-domain none
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: cheetah.pear.relay.mailchannels.net [23.83.216.34] not internal
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: not authenticated
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: DKIM verification successful
mailserver                | Nov 18 10:59:26 mail opendkim[1138]: B0E6F5828BC: s=default d=sender-domain a=rsa-sha256 SSL
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC: ignoring invalid ARC-Authentication-Results header "i=1;#012#011rspamd-55bcb54c45-c8qs5;#012#011auth=pass smtp.auth=thundermail smtp.mailfrom=sender@sender-domain"
mailserver                | Nov 18 10:59:26 mail opendmarc[1114]: B0E6F5828BC ignoring Authentication-Results at 19 from cloud104.unlimitedwebhosting.co.uk
mailserver                | Nov 18 10:59:27 mail opendmarc[1114]: B0E6F5828BC: sender-domain none
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: cheetah.pear.relay.mailchannels.net [23.83.216.34] not internal
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: not authenticated
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: DKIM verification successful
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: s=default d=sender-domain a=rsa-sha256 SSL
mailserver                | Nov 18 10:59:27 mail opendmarc[1114]: B0E6F5828BC: ignoring invalid ARC-Authentication-Results header "i=1;#012#011rspamd-55bcb54c45-c8qs5;#012#011auth=pass smtp.auth=thundermail smtp.mailfrom=sender@sender-domain"
mailserver                | Nov 18 10:59:27 mail opendmarc[1114]: B0E6F5828BC ignoring Authentication-Results at 19 from cloud104.unlimitedwebhosting.co.uk
mailserver                | Nov 18 10:59:27 mail opendmarc[1114]: B0E6F5828BC: sender-domain none
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: cheetah.pear.relay.mailchannels.net [23.83.216.34] not internal
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: not authenticated
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: DKIM verification successful
mailserver                | Nov 18 10:59:27 mail opendkim[1138]: B0E6F5828BC: s=default d=sender-domain a=rsa-sha256 SSL
mailserver                | Nov 18 10:59:27 mail opendmarc[1114]: B0E6F5828BC: ignoring invalid ARC-Authentication-Results header "i=1;#012#011rspamd-55bcb54c45-c8qs5;#012#011auth=pass smtp.auth=thundermail smtp.mailfrom=sender@sender-domain"
mailserver                | Nov 18 10:59:27 mail opendmarc[1114]: B0E6F5828BC ignoring Authentication-Results at 19 from cloud104.unlimitedwebhosting.co.uk
mailserver                | Nov 18 10:59:27 mail opendmarc[1114]: B0E6F5828BC: sender-domain none
mailserver                | Nov 18 10:59:27 mail postfix/qmgr[176941]: B0E6F5828BC: from=<sender@sender-domain>, size=5194, nrcpt=1 (queue active)
mailserver                | Nov 18 10:59:27 mail postfix/smtpd[177011]: disconnect from cheetah.pear.relay.mailchannels.net[23.83.216.34] ehlo=2 starttls=1 mail=1 rcpt=1 data=1 quit=1 commands=7
mailserver                | Nov 18 10:59:27 mail postfix/smtpd-amavis/smtpd[177249]: connect from localhost[127.0.0.1]
mailserver                | Nov 18 10:59:27 mail postfix/smtpd-amavis/smtpd[177249]: E097F5828BD: client=localhost[127.0.0.1]
mailserver                | Nov 18 10:59:27 mail postfix/cleanup[177233]: E097F5828BD: message-id=<20c59ac9-9afc-46bc-afc9-1b4c7763e612@sender-domain>
mailserver                | Nov 18 10:59:27 mail postfix/qmgr[176941]: E097F5828BD: from=<sender@sender-domain>, size=5726, nrcpt=1 (queue active)
mailserver                | Nov 18 10:59:27 mail amavis[176963]: (176963-01) Passed CLEAN {RelayedOpenRelay}, [23.83.216.34]:23397 [senders-mailserver-ip] <sender@sender-domain> -> <me@gmail.com>, Queue-ID: B0E6F5828BC, Message-ID: <20c59ac9-9afc-46bc-afc9-1b4c7763e612@sender-domain>, mail_id: WOMGCnpLlNdK, Hits: -7.11, size: 5494, queued_as: E097F5828BD, 712 ms
mailserver                | Nov 18 10:59:27 mail postfix/smtp-amavis/smtp[177245]: B0E6F5828BC: to=<me@gmail.com>, orig_to=<test@my-domain>, relay=127.0.0.1[127.0.0.1]:10024, delay=2.9, delays=2.2/0.01/0.01/0.71, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as E097F5828BD)
mailserver                | Nov 18 10:59:27 mail postfix/qmgr[176941]: B0E6F5828BC: removed
mailserver                | Nov 18 10:59:28 mail postfix/smtp[177250]: Trusted TLS connection established to gmail-smtp-in.l.google.com[142.250.27.27]:25: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature ECDSA (P-256) server-digest SHA256
mailserver                | Nov 18 10:59:29 mail postfix/smtp[177250]: E097F5828BD: to=<me@gmail.com>, relay=gmail-smtp-in.l.google.com[142.250.27.27]:25, delay=1.6, delays=0/0.01/0.94/0.64, dsn=5.7.1, status=bounced (host gmail-smtp-in.l.google.com[142.250.27.27] said: 550-5.7.1 [my-mailserver-ip      12] Gmail has detected that this message is 550-5.7.1 likely unsolicited mail. To reduce the amount of spam sent to Gmail, 550-5.7.1 this message has been blocked. Please visit 550-5.7.1  https://support.google.com/mail/?p=UnsolicitedMessageError for more 550 5.7.1 information. ay7-20020a056402202700b00546decb7a5esi2365103edb.233 - gsmtp (in reply to end of DATA command))
mailserver                | Nov 18 10:59:29 mail postfix/cleanup[177233]: 7EB625828BC: message-id=<20231118105929.7EB625828BC@mail.macrostep.bg>
mailserver                | Nov 18 10:59:29 mail postfix/bounce[177254]: E097F5828BD: sender non-delivery notification: 7EB625828BC
mailserver                | Nov 18 10:59:29 mail postfix/qmgr[176941]: 7EB625828BC: from=<>, size=8421, nrcpt=1 (queue active)
mailserver                | Nov 18 10:59:29 mail postfix/qmgr[176941]: E097F5828BD: removed
mailserver                | Nov 18 10:59:30 mail postfix/smtp[177250]: Trusted TLS connection established to mail.sender-domain[senders-mailserver-ip]:25: TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits)
mailserver                | Nov 18 10:59:33 mail postfix/smtp[177250]: 7EB625828BC: to=<sender@sender-domain>, relay=mail.sender-domain[senders-mailserver-ip]:25, delay=4.1, delays=0/0/0.73/3.4, dsn=2.0.0, status=sent (250 2.0.0 Ok: queued as 14986C0D5A5)
mailserver                | Nov 18 10:59:33 mail postfix/qmgr[176941]: 7EB625828BC: removed
mailserver                | Nov 18 11:00:04 mail postfix/postscreen[177010]: CONNECT from [194.33.191.162]:63559 to [172.20.0.3]:25
mailserver                | Nov 18 11:00:04 mail postfix/postscreen[177010]: PASS OLD [194.33.191.162]:63559
mailserver                | Nov 18 11:00:04 mail postfix/smtpd[177011]: connect from unknown[194.33.191.162]
mailserver                | Nov 18 11:00:04 mail postfix/smtpd[177011]: Anonymous TLS connection established from unknown[194.33.191.162]: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
mailserver                | Nov 18 11:00:05 mail postfix/smtpd[177011]: lost connection after AUTH from unknown[194.33.191.162]
```

**Comment by Steve-Henley** [comment]
Seems to me the only sure fix is for everyone to stop using Google if they want to be sure of receiving all of the genuine emails sent to them. Their own email client can filter the spam.

**Comment by georglauterbach** [comment]
Did anything change in your setup before these errors started showing up?

**Comment by tzerber** [comment]
There have been no changes in both software and hardware. It just started doing this. I also did a mxtoolbox check against the server - no problems at all.

**Comment by georglauterbach** [comment]
Then this does not seem to be a bug with DMS, right?

**Comment by tzerber** [comment]
> Then this does not seem to be a bug with DMS, right? I am asking because we need to triage this for v13.0.0.

To me DMS trying to present itself as other server is actually a DMS problem(feel free to correct me if I'm wrong on this) and/or inability to forward emails from DMS to Gmail is also a thing.
I spent quite a lot of time looking at the issues here, and i found some old(2017 or so) comment with problem like this but nothing of use to me. I've also read about Google's intentions about the ARK headers and forwarders and so on, but that's planned for next year. As for 13.0.0 - I can swap to edge and try again, but that's a live production mail server that I'd like to avoid running on not-so-stable releases. Normally, i wait at least a month or more before updating just to be safe (unless something breaking requires instant-upgrade). I understand the need of triage but I'm on a dead end here, trying to pick direction where to look for problems and i would be grateful on some help.

**Comment by casperklein** [comment]
Forwarding/sending mails for domains where DMS is not authorized for (SPF etc.) is [problematic](https://www.google.com/search?client=firefox-b-d&q=mail+forwarding+spf+problem).

Did you check `https://support.google.com/mail/?p=UnsolicitedMessageError` as mentioned in the error message? Also check if your server's IP address ended up on blacklists.

**Comment by polarathene** [comment]
> For some unknown reason, when forwarding the mail my mailserver is trying to present itself as the sender's domain.
> The sender's domain is properly set up (dkim,spf etc) and my server is also properly set up (dmarc dkim spf etc).

> To me DMS trying to present itself as other server is actually a DMS problem(feel free to correct me if I'm wrong on this) and/or inability to forward emails from DMS to Gmail is also a thing.

DMS with an alias to send mail outbound IIRC functions similar to sending mail outbound via submission ports. That is DMS should be responsible for the sender domain it claims to be, or at the very least authorized to send on behalf like when configured as a relay.

For example, you could send mail out from DMS with a sender address of `hello@gmail.com` which should be rejected by the MTA of the recipient address because DMS is not authorized for that `gmail.com` domain.

This would be the same with any third-party domain you have delivering mail to DMS on port 25 or a submission port. Alias addresses are checked for matching sender or recipient IIRC, then the delivery process proceeds.
- `alias@example.com` resolves to `hello@gmail.com`.
- Mail is sent outbound to the resolved recipient alias `hello@gmail.com` with whatever sender address was used, but sent via the MTA at `mail.example.com` (DMS FQDN) instead of the original MTA that sent mail to DMS.
- Receiving MTA performs verification that DMS is permitted to send mail on behalf of sender address, otherwise flags/rejects it.

---

DMS can be configured with mail domain DNS:
- MX records pointing to DMS FQDN to direct mail delivery to DMS.
- SPF and friends that authorize DMS FQDN to send mail with sender address on behalf of that domain.
- DKIM public key that DMS should have the appropriate private key to sign outgoing mail with.
- Some users configure DNS records (_and mail accounts_) incorrectly, mixing up the distinction between the DMS host and the separate mail domain(s), which do not need to be related (_`hostname: mail.example.com` is not required to support `hello@example.com`_).

Common setup is to have `mail.example.com` as the FQDN for DMS (_aka `hostname` in `compose.yaml`_), while mail accounts are using `@example.com`. There is some implicit default assumptions from long ago in the codebase that setup DMS with that expectation by default.

Adding accounts to DMS will update the vhosts config, which can affect configuration elsewhere (eg: DKIM key generation).

`mail.example.com` is reserved for internal mail delivery (_assuming aliases are configured, which they're not by default, these differ from virtual aliases that `setup add alias ...`  manages_), specifically for internal services (system users) or mail from Postfix. This mail is likely delivered under `/var/mail`, but not managed by Dovecot (_unless routed to a DMS managed account_). Should be responsible for handling bounces too IIRC.

I recall a common issue with Gmail support was related to IPv6. Unless you need IPv6, ensure that your server / DMS is not reachable with IPv6, otherwise Docker may route an IPv6 connection on the host to an IPv4 address for the DMS container. If you are not familiar with this sort of configuration see the current `:edge` docs, v13 release docs bring a rewrite to IPv6 page with much better guidance.

There is likely a misconfiguration somewhere related to the above, or an incorrect expectation / misunderstanding of a feature being used. I could also be mistaken with the above, but lack an environment setup to verify all that at present.

For forwarding mail, we do have PostSRSd, but I'm not familiar with it enough to assist there 😅

**Comment by tzerber** [comment]
> Forwarding/sending mails for domains where DMS is not authorized for (SPF etc.) is [problematic](https://www.google.com/search?client=firefox-b-d&q=mail+forwarding+spf+problem).

That's where i am at the moment, it fails on this.

> Did you check `https://support.google.com/mail/?p=UnsolicitedMessageError` as mentioned in the error message? Also check if your server's IP address ended up on blacklists.

All checks pass, DMS hosts few email domains with moderate amount of traffic without issue. Forwarding started to fail towards gmail recently.

---
> > For some unknown reason, when forwarding the mail my mailserver is trying to present itself as the sender's domain.
> > The sender's domain is properly set up (dkim,spf etc) and my server is also properly set up (dmarc dkim spf etc).
> 
> > To me DMS trying to present itself as other server is actually a DMS problem(feel free to correct me if I'm wrong on this) and/or inability to forward emails from DMS to Gmail is also a thing.
> 
> DMS with an alias to send mail outbound IIRC functions similar to sending mail outbound via submission ports. That is DMS should be responsible for the sender domain it claims to be, or at the very least authorized to send on behalf like when configured as a relay.
> 
> For example, you could send mail out from DMS with a sender address of `hello@gmail.com` which should be rejected by the MTA of the recipient address because DMS is not authorized for that `gmail.com` domain.
> 
> This would be the same with any third-party domain you have delivering mail to DMS on port 25 or a submission port. Alias addresses are checked for matching sender or recipient IIRC, then the delivery process proceeds.
> 
>     * `alias@example.com` resolves to `hello@gmail.com`.
> 
>     * Mail is sent outbound to the resolved recipient alias `hello@gmail.com` with whatever sender address was used, but sent via the MTA at `mail.example.com` (DMS FQDN) instead of the original MTA that sent mail to DMS.
> 
>     * Receiving MTA performs verification that DMS is permitted to send mail on behalf of sender address, otherwise flags/rejects it.
> 
> 
> DMS can be configured with mail domain DNS:
> 
>     * MX records pointing to DMS FQDN to direct mail delivery to DMS.
> 
>     * SPF and friends that authorize DMS FQDN to send mail with sender address on behalf of that domain.
> 
>     * DKIM public key that DMS should have the appropriate private key to sign outgoing mail with.
> 
>     * Some users configure DNS records (_and mail accounts_) incorrectly, mixing up the distinction between the DMS host and the separate mail domain(s), which do not need to be related (_`hostname: mail.example.com` is not required to support `hello@example.com`_).
> 

As stated before, DKIM/DMARC/SPF are perfectly fine, confirmed by mxtoolbox.com just to be sure. The host has its DNS/MX records and domain names properly set up, and in the original issue I mentioned it worked for more than a year without issues. The host also has a properly set reverse DNS towards the mail IP by the ISP providing the said IP. If someone still has this in question, drop me a pm and I will give you the domain IP and names to check yourself, I just would like to avoid dropping them in on the public eye. 
> 
> Common setup is to have `mail.example.com` as the FQDN for DMS (_aka `hostname` in `compose.yaml`_), while mail accounts are using `@example.com`. There is some implicit default assumptions from long ago in the codebase that setup DMS with that expectation by default.
> 
> Adding accounts to DMS will update the vhosts config, which can affect configuration elsewhere (eg: DKIM key generation).
> 
> `mail.example.com` is reserved for internal mail delivery (_assuming aliases are configured, which they're not by default, these differ from virtual aliases that `setup add alias ...` manages_), specifically for internal services (system users) or mail from Postfix. This mail is likely delivered under `/var/mail`, but not managed by Dovecot (_unless routed to a DMS managed account_). Should be responsible for handling bounces too IIRC.
> 
> I recall a common issue with Gmail support was related to IPv6. Unless you need IPv6, ensure that your server / DMS is not reachable with IPv6, otherwise Docker may route an IPv6 connection on the host to an IPv4 address for the DMS container. If you are not familiar with this sort of configuration see the current `:edge` docs, v13 release docs bring a rewrite to IPv6 page with much better guidance.
> 
> There is likely a misconfiguration somewhere related to the above, or an incorrect expectation / misunderstanding of a feature being used. I could also be mistaken with the above, but lack an environment setup to verify all that at present.
> 

Actually, the configuration is quite okay, will explain it below.

> For forwarding mail, we do have PostSRSd, but I'm not familiar with it enough to assist there 😅

After digging in there for a few hours, I noticed it's not the option you want for for multi-domain setup with both inboxes and forwarders. It works, but only if you're relaying mail. I honestly didn't dig too far, because I don't really want to rewrite the sender address.

---
After another two days of digging around, I think I found the problem but it amazes me that nobody had it before and I believe I found a solution, but keep in mind I'm not an expert in this field. To paint a better example of my issue: 
Random `domain A` sends an email to my `domain B` to the inbox `someone@B` which is set to be an alias to `someone@gmail.com`. 
In normal cases and if  `someone@b` was a normal inbox, Gmail will check SPF and DKIM, if that passes - it passes DMARC as well and mail is delivered. 

In the case of the forwarder: SPF check on the IP of the originating (from gmail's MTA point of view) email server fails, because the connection comes from the DMS ip (`domain B`), whereas in the mail envelope itself, it has headers that came from a different place (`domain A`). From there, the mail directly bounces, based on Big G's policy because if SPF fails, DMARC also fails. (their servers, their policy, can't blame them, there's quite a lot of spam delivered on a daily basis).

Solution:
I stumbled across this thing : https://datatracker.ietf.org/doc/html/rfc8617. A more human-readable format can be found here
https://proton.me/blog/what-is-authenticated-received-chain-arc

Basically, it adds some headers to the mail, confirming that it came from `domain A` but it was relayed by `domain B` and/or any other MTAs on the way to its final destination. I have no idea how this can be done, but the more I read about my issue the more I start to believe I somehow need to implement this in order to be able to forward any mails to Gmail. DMS docs have no mention of these ARC headers at all.

**Comment by polarathene** [comment]
> DMS docs have no mention of these ARC headers at all.

We don't implement them. Contribution is welcome.

I think that is a feature rspamd might be able to assist with.

**Comment by georglauterbach** [comment]
Rspamd is capable of ARC, but I have not had a look yet. We could take a look at this together; that be nice because I am not forwarding e-mails myself. Here is the documentation of the ARC module of Rspamd: <https://rspamd.com/doc/modules/arc.html>.

**Comment by Mygod** [comment]
+1 on ARC support. I think I managed to enable it. While openarc could probably be used instead, rspamd is already installed on latest. Here's roughly what I did:

1. Edit `mailserver.env`.
```env
ENABLE_RSPAMD=1
# disable other modules as recommended
ENABLE_AMAVIS=0
ENABLE_OPENDKIM=0
ENABLE_OPENDMARC=0
ENABLE_POLICYD_SPF=0
```
2. (If needed,) reconfigure DKIM with rspamd: https://docker-mailserver.github.io/docker-mailserver/v13.2/config/best-practices/dkim_dmarc_spf/#generating-keys
3. Create `./docker-data/arc.conf`: (taken from https://github.com/rspamd/rspamd/issues/1993#issuecomment-365352966)
```conf
sign_local = false;
allow_hdrfrom_mismatch = true;
use_domain = "recipient";
```
4. Map the file to enable arc module.
```yaml
services:
  mailserver:
    volumes:
      - ./docker-data/arc.conf:/etc/rspamd/local.d/arc.conf   # CHANGED: Enable arc module in rspamd
```

**Comment by polarathene** [comment]
@Mygod probably best to raise a feature request and provide that information.

I'm fairly confident it'll get acted upon in future once some current tasks settle down.

**Comment by tzerber** [comment]
I can confirm using rspamd works, but I'm still testing (in production :D) for possible problems . While i was exploring options i noticed Openarc is not maintained for a while, but rspamd is both still maintained properly and already installed in DMS.

**Comment by polarathene** [comment]
> Openarc is not maintained for a while

This was also apparently looked into and discussed in DMS issues from 2020 where that was the same impression then.

I don't think we'd see OpenARC support, but it's likely it'd be implemented for DMS with Rspamd equivalent.

**Comment by georglauterbach** [comment]
If it really is just 3 lines of config, then I see absolutely no issue in implementing it in DMS. Could you please open a new issue specifically for ARC with Rspamd please? Makes tracking easier and shorter.
